### PR TITLE
Events

### PR DESCRIPTION
**Pull request by MasumaJaffery**
In this MILESTONE;
1. Add onClick Functions.
2. Import Big.js
3. Implement the Functionality of Calculate and Operate.
Thanks for taking the time to review my PR. 🤓 
Happy Coding!

### CODE REVIEW

File: src/components/Calculator.js [hunk 1]
```diff
@@ -1,47 +1,49 @@
...
-import AppTitle from './AppTitle';
+import CalculatorTitle from './AppTitle';
+import calculate from '../logic/calculate';
 
 function Calculator() {
+  const [display, setdisplay] = useState({
+    total: null,
+    next: null,
+    operation: null,
+  });
```
Comment (Tufahel) [inline src/components/Calculator.js]: - [ ] Well done on implementing `useState`. 👍 But you have missed using props. As one of the main purposes of this project is to learn and use `props` and `useState` so, kindly use `props` with `useState` to grasp the understanding of the previous lessons.
-  Bellow I am giving a suggestion on how you can make the best use of `props` with `useState`.
  - First, you would need to create a JavaScript object containing the calculator's initial state. And you can do it in `App.js`, Pass the data to `/components/Calculator.js`.
  ```
function App() {
    const initialState = {
        total: null,
        next: null,
        operation: null,
      };
  return (
    <div>
        <Calculator initialState={initialState} />
    </div>
  );
}
  ```
  - Then, you can use props `initialState` and replace the `useState` in the `Calculator.js` file like this:
  ```
function Calculator({ initialState }) {
  const [display, setdisplay] = useState(initialState);
  ```

#### NOTE: Here I have just given you an example. It is not must to follow the same way. But use props in the best way you can.

**Review by Tufahel — CHANGES_REQUESTED** [review]
# Status: Required Changes :recycle:
Hi @MasumaJaffery :wave:  

Good job so far! 👍 
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights 🌟

Alt Text

✔️ Descriptive pull request.
✔️ Follow the correct Git flow.
✔️ Professional Readme.
✔️ Nice use of UseState.

## Required Changes ♻️

***Please kindly check the comments/review below.***

## Optional suggestions 🤷‍♂️

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

## Important Notes 📝
- Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
- Please, remember to tag me in your question so I can receive the notification.
- You can mail me at [Mail](mailto:tufahel96@gmail.com) or ask any question at Slack, `Name: Tufahel`

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/components/Calculator.js [hunk 2]
```diff
@@ -1,17 +1,18 @@
-import { useState } from 'react';
+import React, { useState } from 'react';
 import './style.css';
 import CalculatorTitle from './AppTitle';
 import calculate from '../logic/calculate';
 
-function Calculator() {
-  const [display, setdisplay] = useState({
+// eslint-disable-next-line react/prop-types
+function Calculator({ initialState }) {
```
Comment (TadesseDev) [inline src/components/Calculator.js]: - [ ] Good job making the changes as requested by the previous reviewer @MasumaJaffery, but please Consider removing disabled liters. Remember linter is here to force you to write clean code and help you identify what could possibly be improved on the current codebase.

**Review by TadesseDev — CHANGES_REQUESTED** [review]
### Hi @MasumaJaffery 

### I'm Tadesse Alemayehu your reviewer for this PR. :hand: 

Great work on making the changes, but there are some new changes we need to improve based on your new changes 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

199733072-bdacca19-8df9-43a7-be93-3aff2f1c15c8

 ## Highlights
 - [x] Good job making the changes requested by the previous reviewer. 🙏


 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

- N/A


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by sumairq — APPROVED** [review]
### Hi @MasumaJaffery , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
✔️ Good commit history
✔️Linters are passing
✔️Very professional README
✔️Correct Git flow
✔️Descriptive PR
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**